### PR TITLE
Ore Stacking

### PR DESCRIPTION
**Pull request by BlueWildrose**
## About The Pull Request

- All ores have been converted to become stacks.
- Various for loops have been destroyed due to ore stacks now existing.
- Bluespace crystals is detyped from being ores (request of Sili)
- Mining satchels have been recoded to be more like ore boxes (there's an update I'd like to do to them later that will include even bigger ore satchels so I decided this was necessary) and their capacity has been buffed to carrying 200 ores instead of 50. People can now shove up to 7 ore stacks into their backpack with this update, so I decided that made sense.
- .dm file containing ore box code was renamed.
- fixed a bug that allowed you to dupe ores using the ore box ("take" was called twice when inserting ores by hand)
- Some minor polishing of ore box code, particularly examine code, interaction feedback, and allowing you to climb onto boxes.
- Ore boxes when destroyed will drop all of their ores, unless there's too much ore (up to 200 stacks will be dropped). It'll also drop 5 hardwood planks. People who accidentally shoot their box with their PKA, rejoice.
- You can craft ore boxes using 5 hardwood planks.

## Why It's Good For The Game

Having ores be stackable will greatly reduce lag. Otherwise, QoL changes good.

## Changelog

:cl:
add: You can climb ore boxes.
qol: Ore boxes will drop their contents when destroyed. You can also craft ore boxes using hardwood planks now.
balance: Mining satchels can now carry 200 ores instead of 50.
fix: Ore box examines look proper now.
fix: Fixed a bug that allowed you to dupe ores using the ore box.
imageadd: stacks_ore.dmi was added. It holds ore stack stuff.
code: Orebox DM file renamed to ore_box.dm
code: Bluespace crystals are detyped from being ores.
refactor: Ore is now stackable. This will reduce lag when miners are out causing havoc or shoving ore onto conveyor belts.
/:cl:

### CODE REVIEW

File: code/game/objects/items/storage/bags.dm [hunk 1]
```diff
@@ -131,26 +134,30 @@
 /obj/item/storage/bag/ore/gather_all(turf/T as turf, mob/user as mob, var/silent = 0)
 	var/success = 0
 	var/failure = 0
-	for(var/obj/item/ore/I in T) //Only ever grabs ores. Doesn't do any extraneous checks, as all ore is the same size. Tons of checks means it causes hanging for up to three seconds.
-		if(contents.len >= max_storage_space)
+	for(var/obj/item/stack/ore/I in T) //Only ever grabs ores. Doesn't do any extraneous checks, as all ore is the same size. Tons of checks means it causes hanging for up to three seconds.
+		if(istype(user.pulling, /obj/structure/ore_box))
+			var/obj/structure/ore_box/O = user.pulling
+			I.forceMove(O)
+			if(world.time >= last_message == 0 && !silent)
```
Comment (Lohikar) [inline code/game/objects/items/storage/bags.dm:141]: `if(world.time < last_message && !silent)` would be clearer.

File: code/game/objects/items/storage/bags.dm [hunk 2]
```diff
@@ -131,26 +134,30 @@
...
-		success = 1
+		else
+			I.forceMove(src)
+			success = 1
 	if(success && !failure && !silent)
 		if(world.time >= last_message == 0)
 			to_chat(user, "<span class='notice'>You put everything in [src].</span>")
 			last_message = world.time + 10
-	else if(success && (!silent || (silent && contents.len >= max_storage_space)))
+	else if(success && (!silent || (silent && total_ore >= max_ore)))
```
Comment (Lohikar) [inline code/game/objects/items/storage/bags.dm:154]: `else if (success && (!silent || total_ore >= max_ore)`

File: code/modules/mining/ore.dm [hunk 3]
```diff
@@ -1,141 +1,169 @@
...
+/obj/item/stack/ore/verdantium/throw_impact(atom/hit_atom)
 	..()
 	var/mob/living/carbon/human/H = hit_atom
 	if(istype(H) && H.has_eyes() && prob(85))
 		to_chat(H, "<span class='danger'>Some of \the [src] gets in your eyes!</span>")
 		H.Blind(10)
 		H.eye_blurry += 15
 		spawn(1)
-			if(istype(loc, /turf/)) qdel(src)
+			if(istype(loc, /turf/))
```
Comment (Lohikar) [inline code/modules/mining/ore.dm:148]: `isturf(loc)`

File: code/modules/mining/ore_box.dm [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+/// Sigh.
+/obj/structure/ore_box/Entered(atom/movable/AM, atom/oldLoc)
+	. = ..()
+	if(istype(AM, /obj/item/stack/ore))
+		take(AM)
+
+/obj/structure/ore_box/drop_products(method, atom/where)
+	. = ..()
+	var/i = 0
+	while(!isEmpty() && i++ < 200) //may be laggy as hell if they're carrying 2,000+ sand, so capping it at 200 items dropped.
```
Comment (Lohikar) [inline code/modules/mining/ore_box.dm:59]: It's minor, but incrementing `i` in the loop body instead of the control statement would be clearer.

**Review by silicons — APPROVED** [review]

**Pull request merged**